src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/CompilerScope.Storage.cs

```diff
@@ -47,7 +47,9 @@ internal LocalStorage(LambdaCompiler compiler, ParameterExpression variable)
                 // the compiler when emitting an inlined lambda invoke, to
                 // handle ByRef parameters. BlockExpression prevents this
                 // from being exposed to user created trees.
-                _local = compiler.GetNamedLocal(variable.IsByRef ? variable.Type.MakeByRefType() : variable.Type, variable);
+
+                // Set name if DebugInfoGenerator support is brought back.
```

Comment:
Even when DebugInfoGenerator is back we can reuse named local slots when no debug info is emitted.

---
In reply to: [113828475](https://github.com/dotnet/corefx/pull/18564#discussion_r113828475) [](ancestors = 113828475,112038242)